### PR TITLE
Feature/mqtt qos levels

### PR DESCRIPTION
**Pull request by hikhvar**
Implement original proposal of #14.

I think the later proposal with random values per message are not that easy to implement. A normal distribution is easy. However to have more realistic benchmarks one would model their message distribution. Implementing different random distributions is not that easy.

### CODE REVIEW

**Comment by arnoldbechtoldt** [comment]
I'll take a look until Sunday. Thanks for the PR. 👍

**Comment by hikhvar** [comment]
Any feedback?

File: main.go [hunk 1]
```diff
@@ -28,21 +28,23 @@ var (
 	errorLogger   = log.New(os.Stderr, "ERROR: ", log.Lmicroseconds|log.Ltime|log.Lshortfile)
 	verboseLogger = log.New(os.Stderr, "DEBUG: ", log.Lmicroseconds|log.Ltime|log.Lshortfile)
 
-	argNumClients         = flag.Int("num-clients", 10, "Number of concurrent clients")
-	argNumMessages        = flag.Int("num-messages", 10, "Number of messages shipped by client")
-	argTimeout            = flag.String("timeout", "5s", "Timeout for pub/sub actions")
-	argGlobalTimeout      = flag.String("global-timeout", "60s", "Timeout spanning all operations")
-	argRampUpSize         = flag.Int("rampup-size", 100, "Size of rampup batch. Default rampup batch size is 100.")
-	argRampUpDelay        = flag.String("rampup-delay", "500ms", "Time between batch rampups")
-	argTearDownDelay      = flag.String("teardown-delay", "5s", "Graceperiod to complete remaining workers")
```
Comment (arnisoph) [inline main.go]: good catch! 👍

**Comment by arnisoph** [comment]
LGTM! thx for contributing.

**Pull request merged**